src/helm/benchmark/presentation/summarize.py

```diff
@@ -403,20 +403,22 @@ def get_stats_file_metadata(data_overlap_dir: str) -> Dict[str, List[str]]:
                 data_overlap_stats_key = data_overlap_stats.data_overlap_stats_key
                 light_scenario_key = data_overlap_stats_key.light_scenario_key
                 scenario_spec = light_scenario_key.scenario_spec
-                num_instances = data_overlap_stats.num_instances
                 n = data_overlap_stats_key.overlap_protocol_spec.n
-                """
-                TODO: here we are currently just aggregating across all instance ids
-                for a given scenario rather than the subset run on HELM
-                """
-                num_overlapping_inputs = len(data_overlap_stats.instance_ids_with_overlapping_input)
-                num_overlapping_references = len(data_overlap_stats.instance_ids_with_overlapping_reference)
-                if n == OVERLAP_N_COUNT:
-                    scenario_spec_overlap_counts[scenario_spec] = (
-                        num_instances,
-                        num_overlapping_inputs,
-                        num_overlapping_references,
+                if scenario_spec in self.scenario_spec_instance_id_dict:
```

Comment:
Put a comment on what we're doing here.  This is conceptually quite important.

Reply:
Added